multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/PollServiceCreateOrUpdateOperationsExecution.java

```diff
@@ -27,15 +33,14 @@ protected List<CloudServiceInstanceExtended> getServicesData(ProcessContext cont
         // There's no need to poll the creation or update of user-provided services, because it is done synchronously:
         return allServicesToCreate.stream()
                                   .filter(s -> !s.isUserProvided())
-                                  .collect(Collectors.toList());
+                                  .toList();
     }
 
     @Override
     protected ServiceOperation mapOperationState(StepLogger stepLogger, ServiceOperation lastServiceOperation,
                                                  CloudServiceInstanceExtended service) {
         lastServiceOperation = super.mapOperationState(stepLogger, lastServiceOperation, service);
-        // Be fault tolerant on failure on update of service
-        if (lastServiceOperation.getType() == ServiceOperation.Type.UPDATE
+        if ((lastServiceOperation.getType() == ServiceOperation.Type.UPDATE && !shouldFailOnUpdateSupplier.get())
             && lastServiceOperation.getState() == ServiceOperation.State.FAILED) {
```

Comment:
The three conditionals make it a bit hard to read. Consider some way to improve clarity.

Reply:
done